scripts/test_restore_points.sh

```diff
@@ -0,0 +1,302 @@
+max_prepared_transactions = 100
+wal_level = 'archive'
+archive_mode = on
+archive_command = 'test ! -f ${WAL_DIR}/${NAME}/%f && cp %p ${WAL_DIR}/${NAME}/%f'
+restore_command = 'cp ${WAL_DIR}/${NAME}/%f %p'
+recovery_target_name = '${RESTORE_POINT_NAME}'
+recovery_target_action = 'promote'
+EOF
+}
+
+function instance_start()
+{
+	NAME=$1
+	HOST=$2
+	PORT=$3
+	pg_ctl start -D "${STORAGE_DIR}/${NAME}"
+	for i in {1..20}; do
+		sleep 0.5
+		pg_isready -h ${HOST} -p ${PORT} -d ${NAME}
+		if [[ $? == 0 ]] ; then
+			sleep 0.2
+			return 0
+		fi
+	done
+	exit 1
+}
+
+function instance_create()
+{
+	NAME=$1
+	HOST=$2
+	PORT=$3
+	RESTORE_POINT_NAME=$4
+	mkdir -p "${WAL_DIR}/${NAME}"
+	mkdir -p "${STORAGE_DIR}/${NAME}"
+	mkdir -p "${BASEBACKUP_DIR}/${NAME}"
+	instance_configure ${NAME} ${PORT} ${RESTORE_POINT_NAME}
+	instance_start ${NAME} ${HOST} ${PORT}
+}
+
+function instance_stop()
+{
+	NAME=$1
+	pg_ctl stop -D "${STORAGE_DIR}/${NAME}"
+}
+
+function instance_recover()
+{
+	NAME=$1
+	HOST=$2
+	PORT=$3
+	cp -R "${BASEBACKUP_DIR}/${NAME}" "${STORAGE_DIR}/${NAME}"
+	chmod 700 "${STORAGE_DIR}/${NAME}"
+	touch "${STORAGE_DIR}/${NAME}/recovery.signal"
+	instance_start ${NAME} ${HOST} ${PORT}
+}
+
+function instance_basebackup()
+{
+	NAME=$1
+	HOST=$2
+	PORT=$3
+	echo "CHECKPOINT" | psql -p ${PORT} -h ${HOST} ${NAME}
+	pg_basebackup --progress -p ${PORT} -h ${HOST} -D "${BASEBACKUP_DIR}/${NAME}"
+}
+
+function instance_cleanup_storage()
+{
+	NAME=$1
+	rm -rf "${STORAGE_DIR}/${NAME}"
+}
+
+function instance_cleanup()
+{
+	NAME=$1
+	rm -rf "${STORAGE_DIR}/${NAME}"
+	rm -rf "${WAL_DIR}/${NAME}"
+	rm -rf "${BASEBACKUP_DIR}/${NAME}"
+}
+
+function instance_cleanup_dirs()
+{
+	rm -rf ${STORAGE_DIR}
+	rm -rf ${WAL_DIR}
+	rm -rf ${BASEBACKUP_DIR}
+	rm -rf ${RESULT_DIR}
+}
+
+function test_single_node()
+{
+	RESTORE_POINT_NAME=$1
+	RESULT_FILE_NAME=${RESULT_DIR}/single_node_${RESTORE_POINT_NAME}
+
+	# create and start single node instance
+	instance_create ${SN_NAME} ${HOST} ${SN_PORT} ${RESTORE_POINT_NAME}
+
+	# setup database
+	createdb -p ${SN_PORT} -h ${HOST} ${SN_NAME}
+
+	psql -p ${SN_PORT} -h ${HOST} ${SN_NAME} <<EOF
+CREATE EXTENSION timescaledb;
+
+CREATE TABLE test(time timestamp NOT NULL,  device int,  temp float);
+SELECT create_hypertable('test', 'time', 'device', 3);
+
+INSERT INTO test VALUES('2017-01-01 06:01', 1, 1.1), ('2017-01-01 09:11', 3, 2.1);
+EOF
+	# do initial basebackup
+	instance_basebackup ${SN_NAME} ${HOST} ${SN_PORT}
+
+	# insert more data and create the restore points
+	psql -p ${SN_PORT} -h ${HOST} ${SN_NAME} <<EOF
+SELECT pg_create_restore_point('${TEST_RESTORE_POINT_1}');
+
+INSERT INTO test VALUES('2017-01-01 08:01', 1, 1.2), ('2017-01-02 08:01', 2, 1.3);
+
+SELECT pg_create_restore_point('${TEST_RESTORE_POINT_2}');
+
+INSERT INTO test VALUES
+('2018-07-02 08:01', 87, 1.6),
+('2018-07-01 06:01', 13, 1.4),
+('2018-07-01 09:11', 90, 2.7),
+('2018-07-01 08:01', 29, 1.5);
+
+SELECT pg_create_restore_point('${TEST_RESTORE_POINT_3}');
```

Comment:
Would be good to also insert date after the restore point to see that this is not included in the restored instance.

Reply:
I'll do it, but that's also checked when restoring from the restore points 1 and 2